pdns/dnsdistdist/dnsdist-kvs.cc

```diff
@@ -114,6 +121,51 @@ bool LMDBKVStore::keyExists(const std::string& key)
   return false;
 }
 
+bool LMDBKVStore::getRangeValue(const std::string& key, std::string& value)
+{
+  try {
+    auto transaction = d_env.getROTransaction();
+    auto cursor = transaction->getROCursor(d_dbi);
+    MDBOutVal actualKey;
+    MDBOutVal result;
+    // for range-based lookups, we expect the data in LMDB
+    // to be stored with the last value of the range as key
+    // and the first value of the range as data, sometimes
+    // followed by any other content we don't care about
+
+    // retrieve the first key greater or equal to our key
+    int rc = cursor.lower_bound(MDBInVal(key), actualKey, result);
+
+    if (rc == 0) {
+      auto last = actualKey.get<std::string>();
+      if (last.size() != key.size() || key > last) {
```

Comment:
hmmm, do string compares work for abbreviated addresses? Likely not, so it should be documented that this only works for fully written out v6 addresses.

Likewise I do not think string compare works for IPv4 in their normal presentation form: take for example `200.1.1.1` which sorts before `3.1.1.1` or `1.1.1.200` which sort before `1.1.1.3`
We probably should document that v4 string should be represented with leading zeroes (though watch out: that could lead to octets being interpreted as octal in some contexts).

Reply:
The logic assumes that addresses are stored in binary, in network order, as returned by `KeyValueLookupKeySourceIP`. I guess I need to document that more clearly.

Reply:
Right, I made a wrong assumption, as long as embedded NUL bytes are handled ok that should be fine indeed.

Reply:
You should not have had to guess, so I documented that a bit better :)